src/safeposix/syscalls/fs_calls.rs

```diff
@@ -126,39 +181,57 @@ impl Cage {
                 }
                 let size;
 
+                // Fetch the Inode Object associated with the inode number of the existing file.
+                // There are different Inode types supported by the open_syscall (i.e., File, Directory, Socket, CharDev).
                 let mut inodeobj = FS_METADATA.inodetable.get_mut(&inodenum).unwrap();
                 match *inodeobj {
                     Inode::File(ref mut f) => {
+                        //This is a special case when the input flags contain "O_TRUNC" flag, 
+                        //This flag truncates the file size to 0, and the mode and owner are unchanged
+                        // and is only used when the file exists and is a regular file
                         if O_TRUNC == (flags & O_TRUNC) {
-                            // We only do this to regular files, otherwise O_TRUNC is undefined
-                            //close the file object if another cage has it open
+                            // Close the existing file object and remove it from the FileObject Hashtable using the inodenumber
                             let entry = FILEOBJECTTABLE.entry(inodenum);
                             if let interface::RustHashEntry::Occupied(occ) = &entry {
+                                // Get the entry for the current file associated with the inodeNumber and close the opened it
```

Comment:
typo in comment.

Also, why do we close it?  What is the reason?

Reply:
In the current implementation, I can see that the FileObjectTable contains entries about the `open` files in the system. So, when the TRUNC flag is set, we close the existing opened file and remove its entry from the FileObjectTable. Then the size is set to 0, file is inserted back to the FileObjectTable in the open state.